server/app/controllers/admin/AdminImportController.java

```diff
@@ -145,6 +152,41 @@ public Result hxImportProgram(Http.Request request) {
               .render());
     }
 
+    // Prevent admin from importing a program with visiblity set to "Visible to selected trusted
+    // intermediaries only" since we don't migrate TI groups
+    if (program.displayMode() == DisplayMode.SELECT_TI) {
+      return ok(
+          adminImportViewPartial
+              .renderError(
+                  "Display mode 'SELECT_TI' is not allowed.",
+                  "Please select another program display mode and try again")
+              .render());
+    }
+
+    // Check for other validation errors like invalid program admin names
+    ImmutableList<String> notificationPreferences =
+        program.notificationPreferences().stream()
+            .map(preference -> preference.getValue())
+            .collect(ImmutableList.toImmutableList());
+    ImmutableSet<CiviFormError> programErrors =
+        programService.validateProgramDataForCreate(
+            program.adminName(),
+            program.localizedName().getDefault(),
+            program.localizedDescription().getDefault(),
+            program.externalLink(),
+            program.displayMode().getValue(),
+            notificationPreferences,
+            ImmutableList.of(), // categories are not migrated
+            ImmutableList.of() // associated TI groups are not migrated
+            );
+    if (!programErrors.isEmpty()) {
+      String errorString = joinErrors(programErrors).replace("URL", "admin name");
```

Comment:
When submitting a new or edited program through the program UI the errors reference "URL" which makes sense. But here saying "URL" would be confusing since admin will need to update the program admin name, not the url, to fix errors.

Reply:
It may be worth adding this (or something like it) as a comment above the code as well :)

Reply:
I was thinking the same thing! Done.